src/Async/Async.spec.js

```diff
@@ -4,7 +4,7 @@ const helpers = require('../test/helpers')
 
 const bindFunc = helpers.bindFunc
 
-const List = require('../core/List')
+const List = require('../List')
```

Comment:
So this is interesting, cannot believe I missed this in the original definition. Sorry to add more work to this.

There are a couple things here that will need to be changed with the `Async` Type.
First of all in `Async`, for `Async.all` we check for a foldable type, and we also need to check with `isTraversable`. Which means that `isTraversable` will then also need to be moved core folder so it can be referenced in `Async`
Then on top of that, `sequence` is not defined by the fantasy-land spec, so we would need to change the `all` implementation to be: `asyncs.traverse(Async.of, x => x)`, but `array` can stay as is.

Then after all of that, the spec for `Async` for `all` working on Lists, should work with any Traversable containing Async. then the [`MockCrock`](https://github.com/evilsoft/crocks/blob/master/src/test/MockCrock.js) can be in place of `List` for the `all` spec.

So basically the following needs to happen:
* Move functionality for `isTraversable` into core.
* Update `isTraversable` in predicates to use core version.
* In `Async.all` implementation, add a check for `isTraversable` before traversing.
* update error to mention `Traversable` as well as foldable.
* change from sequence to traverse for the non-array case.
* update the `VERSION` of `Async` as this is an api change.
* update all spec in Async to use the `MockCrock`

So the implementation of Async.all should look like:
```js
const allErr = 'Async.all: Foldable Traversable structure of Asyncs required'
function all(asyncs) {
  if(!(isFoldable(asyncs) && allAsyncs(asyncs))) {
    throw new TypeError(allErr)
  }

  if(isArray(asyncs)) {
    return array.sequence(Async.of, asyncs)
  }

  if(isTraverable(asyncs)) {
    return asyncs.traverse(Async.of, x => x)
  }

  throw new TypeError(allErr)
}
```

Reply:
Sounds doable.

Reply:
Still sounds doable but I don't believe I have time in the short term future to do this. Any takers?